runtime/memory/64bit_nix.c

```diff
@@ -52,10 +52,13 @@ i32 instruction_memory_grow(i32 count) {
     return prev_size;
 }
 
+INLINE void check_bounds(u32 offset, u32 bounds_check) {
+    // Due to how we setup memory for x86, the virtual memory mechanism will catch the error, if bounds < WASM_PAGE_SIZE
+    awsm_assert(bounds_check < WASM_PAGE_SIZE || (memory_size > bounds_check && offset <= memory_size - bounds_check));
+}
```

Comment:
For 64 bit, don't we want to avoid bounds checks? Thought that was the point of the 64 bit version where we can allocate 4GiB. This does not block the PR. Just for discussion after you apply.

Reply:
WASM can address the last byte of memory, which means you need an extra guard page (in case you try and do something like `u64* x = U32_MAX; *x`). This guard page approach only works for free for loadsd that are smaller than a page, for obvious reasons